File: MATLAB/RMS.m
%% Root-mean-square
function [rms] = RMS(audio)
% Spectrum
[xn, ~] = audioread(audio);
xn = (xn(:, 1) + xn(:, 2)) / size(xn, 2);
N = length(xn);
Xk = abs(fft(xn));
Xk = Xk(1 : end/2);

% RMS
rms = sqrt(sum(Xk.^2) / N);

File: Python/Descriptors/RMS.py
import numpy as np

def getRMS(inAudio, unit="db"):
    audio = np.array(inAudio)
    rms = np.sqrt(np.mean(audio**2))

    if unit == "dB":
        decibels = convertToDecibel(rms)
        return decibels
    elif unit == "linear":
        return rms

def convertToDecibel(inAudio):
    ref = 1
    if inAudio != 0:
        return 20 * np.log10(abs(inAudio) / ref)
    else:
        return -60